src/taipy/core/notification/event.py

```diff
@@ -58,6 +60,7 @@ class EventEntityType(_ReprEnum):
 }
 
 
+@dataclass(frozen=True)
```

Comment:
*Note to reviewers*: the use of dataclass makes the definition of this (read-only) class sleaker. It also provides for free default `repr` and `hash`.